src/main/java/org/springframework/retry/backoff/UniformRandomBackOffPolicy.java

```diff
@@ -107,9 +132,10 @@ public long getMaxBackOffPeriod() {
 	 */
 	protected void doBackOff() throws BackOffInterruptedException {
 		try {
-			long delta = maxBackOffPeriod == minBackOffPeriod ? 0
-					: random.nextInt((int) (maxBackOffPeriod - minBackOffPeriod));
-			sleeper.sleep(minBackOffPeriod + delta);
+			Long min = this.minBackOffPeriod.get();
+			long delta = this.maxBackOffPeriod == this.minBackOffPeriod ? 0
```

Comment:
I think this is something wrong: you compare `Supplier` instances - not their values.

Reply:
Oops.